src/tests/unit/tests/background/tab-controller.test.ts

```diff
@@ -653,11 +653,80 @@ describe('TabControllerTest', () => {
         mockChromeAdapter.verifyAll();
     });
 
-    test('onUpdateTab', () => {
-        mockChromeAdapter.setup(ca => ca.addListenerToTabsOnUpdated(It.isAny())).verifiable(Times.once());
+    test('tab change test: url was changed', () => {
+        let tabUpdatedCallback: (tabId: number, changeInfo: chrome.tabs.TabChangeInfo) => void = null;
+        let getTabCallback: (tab: chrome.tabs.Tab) => void;
+        let onReject;
+        const tabId = 1;
+        const getTabCallbackInput = {
+            title: 'new title',
+            url: 'new url',
+            id: tabId,
+        };
 
+        const interpretInput: Message = {
+            messageType: Messages.Tab.Change,
+            payload: getTabCallbackInput,
+            tabId: tabId,
+        };
+        const interpreterMock = Mock.ofType(Interpreter);
+        interpreterMock.setup(im => im.interpret(It.isValue(interpretInput))).verifiable(Times.once());
+        tabInterpreterMap = {
+            1: {
+                interpreter: interpreterMock.object,
+                stores: null,
+            },
+        };
+
+        mockChromeAdapter
+            .setup(ca => ca.addListenerToTabsOnUpdated(It.isAny()))
+            .callback(cb => {
+                tabUpdatedCallback = cb;
+            })
+            .verifiable(Times.once());
+
+        mockChromeAdapter
+            .setup(mca => mca.getTab(It.isValue(tabId), It.isAny(), It.isAny()))
+            .returns((id, cb, reject) => {
+                getTabCallback = cb;
+                onReject = reject;
+            })
+            .verifiable(Times.once());
+        logMock.setup(log => log(`changed tab with Id ${tabId} not found`)).verifiable(Times.once());
+
+        testSubject = createTabControllerWithoutFeatureFlag(tabInterpreterMap);
+        testSubject.initialize();
+        tabUpdatedCallback(tabId, { url: 'some url' });
+        getTabCallback(getTabCallbackInput as any);
```

Comment:
Does Chrome actually have a case where it will invoke both the success *and* the reject callback in response to a single getTab call? Would it make more sense for "call getTabCallback" and "call onReject" cases to be split out into separate test cases?